src/OrchardCore/Orchard.DisplayManagement/Fluid/FluidViewTemplate.cs

```diff
@@ -0,0 +1,212 @@
+
+                    if (fileInfo != null && fileInfo.Exists)
+                    {
+                        viewImportsTemplate = Parse(location, fileProviderAccessor.SharedFileProvider, Cache);
+                        break;
+                    }
+                }
+            }
+
+            path = Path.ChangeExtension(path, ViewExtension);
+            var template = Parse(path, fileProvider, cache);
+
+            var context = new TemplateContext();
+            context.AmbientValues.Add("FluidPage", page);
+
+            var site = await page.GetService<ISiteService>().GetSiteSettingsAsync();
+            context.MemberAccessStrategy.Register(site.GetType());
+            context.LocalScope.SetValue("Site", site);
+
+            context.MemberAccessStrategy.Register(page.Context.GetType());
+            context.MemberAccessStrategy.Register(page.Context.Request.GetType());
+            context.LocalScope.SetValue("Context", page.Context);
+
+            context.LocalScope.SetValue("ViewData", page.ViewData);
+            context.LocalScope.SetValue("ViewContext", page.ViewContext);
+
+            var modelState = page.ViewContext.ModelState.ToDictionary(
+                kv => kv.Key, kv => (object)new ModelStateNode(kv.Value));
+
+            modelState["IsValid"] = page.ViewContext.ModelState.IsValid;
+            context.LocalScope.SetValue("ModelState", modelState);
+
+            context.RegisterObject((object)page.Model);
+            context.LocalScope.SetValue("Model", page.Model);
+
+            if (page.Model is Shape)
+            {
+                if (page.Model.Items.Count > 0)
+                {
+                    foreach (var item in page.Model.Items)
+                    {
+                        context.RegisterObject((object)item);
+                    }
+                }
+
+                if (page.Model.Properties.Count > 0)
+                {
+                    foreach (var prop in page.Model.Properties)
+                    {
+                        context.RegisterObject((object)prop.Value);
+                        context.LocalScope.SetValue(prop.Key, prop.Value);
+                    }
+                }
+            }
+
+            if (page.Model.Field != null)
+            {
+                context.RegisterObject(((object)page.Model.Field));
+            }
+
+            if (viewImportsTemplate != null)
+            {
+                await viewImportsTemplate.RenderAsync(context);
+            }
+
+            page.WriteLiteral(await template.RenderAsync(context));
+        }
+
+        internal static IFluidTemplate Parse(string path, IFileProvider fileProvider, IMemoryCache cache)
+        {
+            return cache.GetOrCreate(path, entry =>
+            {
+                entry.SlidingExpiration = TimeSpan.FromHours(1);
+
+                var fileInfo = fileProvider.GetFileInfo(path);
+                entry.ExpirationTokens.Add(fileProvider.Watch(path));
```

Comment:
Anyway i think we don't need it because `ViewImport` has its own `ChangeToken` so when updated it is re-parsed. And this is when we do `RenderAsync()` that all needed statements have to be executed.

Just tried to remove a line in `_ViewImports.liquid`, then i started the web app, then some contents were not rendered. Then i re-added the missing line in `_ViewImports.liquid`, then i did a refresh and all was correctly rendered.